src/web/api/src/main/java/org/locationtech/geogig/web/api/CommandBuilder.java

```diff
@@ -55,471 +88,21 @@
-        } else if ("featurediff".equalsIgnoreCase(commandName)) {
-            command = buildFeatureDiff(options);
-        } else if ("getCommitGraph".equalsIgnoreCase(commandName)) {
-            command = buildGetCommitGraph(options);
-        } else if ("merge".equalsIgnoreCase(commandName)) {
-            command = buildMerge(options);
-        } else if ("checkout".equalsIgnoreCase(commandName)) {
-            command = buildCheckout(options);
-        } else if ("beginTransaction".equalsIgnoreCase(commandName)) {
-            command = buildBeginTransaction(options);
-        } else if ("endTransaction".equalsIgnoreCase(commandName)) {
-            command = buildEndTransaction(options);
-        } else if ("add".equalsIgnoreCase(commandName)) {
-            command = buildAdd(options);
-        } else if ("remove".equalsIgnoreCase(commandName)) {
-            command = buildRemove(options);
-        } else if ("resolveconflict".equalsIgnoreCase(commandName)) {
-            command = buildResolveConflict(options);
-        } else if ("revertfeature".equalsIgnoreCase(commandName)) {
-            command = buildRevertFeature(options);
-        } else if ("rebuildgraph".equalsIgnoreCase(commandName)) {
-            command = buildRebuildGraph(options);
-        } else if ("blame".equalsIgnoreCase(commandName)) {
-            command = buildBlame(options);
-        } else if ("version".equalsIgnoreCase(commandName)) {
-            command = buildVersion(options);
-        } else if ("cat".equalsIgnoreCase(commandName)) {
-            command = buildCat(options);
-        } else if ("statistics".equalsIgnoreCase(commandName)) {
-            command = buildStatistics(options);
-        } else {
+
+        if (!MAPPINGS.containsKey(commandName)) {
             throw new CommandSpecException("'" + commandName + "' is not a geogig command");
         }
 
-        command.setTransactionId(options.getFirstValue("transactionId", null));
-
-        return command;
-    }
+        WebAPICommand command = MAPPINGS.get(commandName).apply(options);
 
-    /**
-     * Parses a string to an Integer, using a default value if the was not found in the parameter
-     * set.
-     * 
-     * @param form the parameter set
-     * @param key the attribute key
-     * @param defaultValue the default value
-     * @return the parsed integer
-     */
-    static Integer parseInt(ParameterSet form, String key, Integer defaultValue) {
-        String val = form.getFirstValue(key);
-        Integer retval = defaultValue;
-        if (val != null) {
-            try {
-                retval = new Integer(val);
-            } catch (NumberFormatException nfe) {
-                throw new CommandSpecException("Invalid value '" + val + "' specified for option: "
-                        + key);
-            }
+        if (command instanceof AbstractWebAPICommand) {
+            ((AbstractWebAPICommand) command).setTransactionId(options.getFirstValue(
```

Comment:
@jdgarrett @groldan Does it make more sense to move **setTransactionId** and **getTransactionId** to the WebAPICommand interface, so we can avoid calling **instance of** here?